components/SwipeRow.js

```diff
@@ -347,27 +423,68 @@ class SwipeRow extends Component {
                 if (
                     this.currentTranslateX - projectedExtraPixels <
                     this.props.rightOpenValue *
-                        (this.props.swipeToOpenPercent / 100)
+                    (this.props.swipeToOpenPercent / 100)
                 ) {
                     // we're more than halfway
                     toValue = this.isForceClosing
                         ? 0
                         : this.props.rightOpenValue;
                 }
+                if (
+                    this.currentTranslateX - projectedExtraPixels <
+                    this.props.rightActivationValue
+                ) {
+                    // we're more than halfway
+                    toValue = this.isForceClosing
+                        ? 0
+                        : this.props.rightActionValue;
+
+                    actionSide = 'right';
+                }
             } else {
                 if (
                     this.currentTranslateX - projectedExtraPixels <
-                    this.props.rightOpenValue *
-                        (1 - this.props.swipeToClosePercent / 100)
+                    this.props.rightOpenValue
                 ) {
                     toValue = this.isForceClosing
                         ? 0
                         : this.props.rightOpenValue;
                 }
+                if (
+                    this.currentTranslateX - projectedExtraPixels <
+                    this.props.rightActivationValue *
+                    (1 - this.props.swipeToClosePercent / 100)
+                ) {
+                    toValue = this.isForceClosing
+                        ? 0
+                        : this.props.rightActionValue;
+
+                    actionSide = 'right';
+                }
             }
         }
 
-        this.manuallySwipeRow(toValue);
+        let action;
+        if (actionSide === 'right') {
+            action = () => {
+                this.props.onRightAction && this.props.onRightAction();
+                !this.state.rightActionEvaluated &&
+                    this.setState({
+                        rightActionEvaluated: true,
+                    });
+            };
+        }
+        if (actionSide === 'left') {
+            action = () => {
+                this.props.onLeftAction && this.props.onLeftAction();
+                !this.state.leftActionEvaluated &&
+                    this.setState({
+                        leftActionEvaluated: true,
+                    });
+            };
+        }
+
+        this.manuallySwipeRow(toValue, action);
```

Comment:
@jemise111, the default animation type is `spring` now... What do you think about the ability to set the type of animation and its settings as props (i.e. animationType, animationConfig)?

Reply:
Absolutely, as long as you keep the defaults as they are now I'm on board

Reply:
I will propose this changes in separate PR